ui/src/services/graphUtils.js

```diff
@@ -205,3 +302,80 @@ export const getNodesInterfacesString = (nodeIPsInfo): [] => {
   const uniqueInterfaces = [...new Set(interfaces)];
   return uniqueInterfaces;
 };
+
+export function renderTooltipSerie({
+  color,
+  isLineDashed,
+  name,
+  value,
+  key,
+}: {
+  color?: string,
+  isLineDashed?: boolean,
+  name: string,
+  value: string,
+  key: string,
+}) {
+  return `<tr>
+    <td class="color" style="width: 1rem;">
+    ${
+      color !== undefined
+        ? `<span style='background: ${
+            isLineDashed
+              ? `repeating-linear-gradient(to right,${color} 0,${color} ${spacing.sp1},transparent ${spacing.sp1},transparent ${spacing.sp2})`
+              : color
+          };width: ${spacing.sp8};height:${
+            spacing.sp2
+          };display: inline-block;vertical-align: middle;'></span>`
+        : ''
+    }
+    </td>
+    <td style="text-align: left;">
+        ${name}
+    </td>
+    <td class="value" style="text-align: right;min-width: 5rem;display: table-cell;">
+      ${value}
+    </td>
+  </tr>`;
+}
+
+export const renderQuantileData = (
+  isIdle,
+  isLoading,
+  isSuccess,
+  data,
+  nodeMapPerIp,
+  theme,
+  valueBase,
+  unitLabel,
+) => {
+  const hoverQuantileValue = (data) => {
+    return unitLabel
+      ? `${parseFloat(data.value[1] / (valueBase || 1)).toFixed(
+          2,
+        )} ${unitLabel}`
+      : `${parseFloat(data.value[1] / (valueBase || 1)).toFixed(2)}`;
+  };
+
+  return `${
+    isIdle || isLoading
+      ? `<tr style="color: ${theme.textSecondary}"><td></td><td colspan='2' style="padding-left: 1rem;">Loading...</td></tr>`
+      : ''
+  }
+  ${
+    isSuccess
+      ? data?.data?.result
```

Comment:
I think we should display an inline message to the user if we fail to load the data for some reason

Reply:
Add `Sorry, failed to load data due to error occurs.` when isSuccess is false